runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -683,121 +684,39 @@ Java_jdk_internal_misc_Unsafe_shouldBeInitialized(JNIEnv *env, jobject receiver,
 	return JNI_FALSE;
 }
 
-void
-registerJdkInternalMiscUnsafeNatives(JNIEnv *env, jclass clazz) {
-	/* clazz can't be null */
-	JNINativeMethod natives[] = {
-		{
-			(char*)"defineClass",
```

Comment:
The replacement set of natives don't match those required by Java 8.
Each of these should omit the `0` suffix for Java 8 (so they can't be classified as 'common'):
```
allocateMemory0
copyMemory0
defineAnonymousClass0
defineClass0
ensureClassInitialized0
freeMemory0
getLoadAverage0
objectFieldOffset0
reallocateMemory0
setMemory0
staticFieldBase0
staticFieldOffset0
```
These don't appear to exist in Java 8 (I would have expected the call to `RegisterNatives` to complain): they should be removed for Java 8:
```
getUncompressedObject
isBigEndian0
unalignedAccess0
```

Reply:
Pls note the comments for the method `registerJdkInternalMiscUnsafeNativesCommon` as following:
/* register jdk.internal.misc.Unsafe natives common to Java 9, 18.3 and beyond */
`Java 8` doesn't have class j.i.m.Unsafe.

Reply:
Java 8 _does_ have sun.misc.Unsafe and it was my understanding that this was how those native were registered (and would be broken by this change). Am I missing something?

Reply:
`Java 8` `s.m.Unsafe` natives are exported explicitly (see `sun_misc_Unsafe_exports.xml`).
Again, `Java 8` doesn't have class j.i.m.Unsafe, whatever change within `Java_jdk_internal_misc_Unsafe_registerNatives` won't affect `Java 8` behaviours.